server/bin/pbench-unpack-tarballs.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+from pathlib import Path
+import sys
+
+from pbench.common.exceptions import BadConfig
+from pbench.common.logger import get_pbench_logger
+from pbench.server import PbenchServerConfig
+from pbench.server.database import init_db
+from pbench.server.unpack_tarballs import Unpack
+
+_NAME_ = "pbench-unpack-tarballs"
+
+# The link source for the operation of this script.
+_linksrc = "TO-UNPACK"
+
+
+def main(cfg_name):
+    if not cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable or use --config <file> on the"
+            " command line",
+            file=sys.stderr,
+        )
+        return 2
+
+    try:
+        config = PbenchServerConfig(cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    # We're going to need the Postgres DB to track dataset state, so setup
+    # DB access.
+    init_db(config, logger)
+
+    BUCKET = str(sys.argv[1])
+    prog = Path(sys.argv[0]).name
+
+    lowerbound, upperbound = 0, float("inf")
+
+    if BUCKET:
+        lowerbound = config.get(f"pbench-unpack-tarballs/{BUCKET}", "lowerbound")
+        if lowerbound:
+            lowerbound = lowerbound * 1024 * 1024
+        upperbound = config.get(f"pbench-unpack-tarballs/{BUCKET}", "upperbound")
+        if upperbound:
+            upperbound = upperbound * 1024 * 1024
+        prog = f"{prog}-{BUCKET}"
+
+    # Initiate the unpacking
+    unpack_obj = Unpack(config, logger)
+    result = unpack_obj.unpack_tarball(lowerbound, upperbound)
```

Comment:
Shouldn't this be plural: `unpack_obj.unpack_tarballs` ? It is potentially unpacling a whole slew.

Reply:
And I think there is a name mismatch anyway: the Unpack class calls the method `unpack`.